supertokens_python/recipe/emailverification/interfaces.py

```diff
@@ -133,59 +126,40 @@ def __init__(self, request: BaseRequest, response: BaseResponse, recipe_id: str,
         self.recipe_implementation = recipe_implementation
 
 
-class EmailVerifyPostResponse(ABC):
-    def __init__(
-            self, status: Literal['OK', 'EMAIL_VERIFICATION_INVALID_TOKEN_ERROR'], user: Union[User, None]):
-        self.status: Literal['OK', 'EMAIL_VERIFICATION_INVALID_TOKEN_ERROR'] = status
-        self.is_ok: bool = False
-        self.is_email_verification_invalid_token_error: bool = False
-        self.user: Union[User, None] = user
-
-    def to_json(self) -> Dict[str, Any]:
-        return {
-            'status': self.status
-        }
-
+class EmailVerifyPostOkResponse(APIResponse):
+    """
+    EmailVerifyPostOkResponse: used to send the ok response if the user
+    is successfully verified.
+    """
 
-class EmailVerifyPostOkResponse(EmailVerifyPostResponse):
     def __init__(self, user: User):
-        super().__init__('OK', user)
-        self.is_ok = True
-        self.is_email_verification_invalid_token_error = False
+        self.user = user
 
     def to_json(self) -> Dict[str, Any]:
-        if self.user is None:
-            raise Exception("Should never come here")
         return {
-            'status': self.status,
+            'status': "OK",
             'user': {
                 'id': self.user.user_id,
                 'email': self.user.email
             }
         }
 
 
-class EmailVerifyPostInvalidTokenErrorResponse(EmailVerifyPostResponse):
-    def __init__(self):
-        super().__init__('EMAIL_VERIFICATION_INVALID_TOKEN_ERROR', None)
-        self.is_ok = False
-        self.is_email_verification_invalid_token_error = True
-
-
-class IsEmailVerifiedGetResponse(ABC):
-    def __init__(self, status: Literal['OK']):
-        self.status = status
-        self.is_ok = False
+class EmailVerifyPostInvalidTokenErrorResponse(APIResponse):
+    """
+    EmailVerifyPostInvalidTokenErrorResponse: used to send the error response if the
+    user varification fails due to invalid token.
+    """
 
     def to_json(self) -> Dict[str, Any]:
         return {
-            'status': self.status
+            'status': 'EMAIL_VERIFICATION_INVALID_TOKEN_ERROR'
         }
 
 
-class IsEmailVerifiedGetOkResponse(IsEmailVerifiedGetResponse):
+class IsEmailVerifiedGetOkResponse(APIResponse):
     def __init__(self, is_verified: bool):
-        super().__init__('OK')
+        self.status = 'OK'
```

Comment:
Be consistent with your changes. In other API return classes, you are not assigning a status field. So why do it here?

Reply:
done.